### ISSUE TITLE
Add "Set" attribute to Cards

### ISSUE THREAD

**Issue by antonioalonzi** [MatagTheGame/matag-cards#5]
This is, "Core Set 2020", "Unglue", "25 Masters"...

**Comment by antonioalonzi** [comment]
@allentiak Moved the card to the `matag-cards` project.

I'm not sure what you are looking for here.
At the moment I already have the concept of [MtgSet](https://github.com/MatagTheGame/matag-cards/blob/master/src/main/java/com/matag/cards/sets/MtgSet.java).


You can get a list of cards contained in a set [see MtgSetsTest](https://github.com/MatagTheGame/matag-cards/blob/master/src/test/java/com/matag/cards/sets/MtgSetsTest.java#L38)


But a card does not have a `sets` attribute where you can say `getSets()` and return a list of all the sets the card was published.

I feel the most natural connection is the opposite `set.getCards()` and I would keep this as source of truth.
I can definitely implement a helper that does the reverse and given a card returns the list of sets that card belongs to.

Is that what you need?

If so I can add methods to [MtgSets](https://github.com/MatagTheGame/matag-cards/blob/master/src/main/java/com/matag/cards/sets/MtgSets.java):
 - getSetsForCard(String name);
 - getSetsForCard(Card card);

**Comment by allentiak** [comment]
@antonioalonzi 

Ciao, Antonio,

Sorry for the late reply.

Thanks for the consideration. However, I think I opened the wrong bug...

In short, I think your data model is inaccurate. And that if you rely on an inaccurate data model, things *will* fail.

I suggest you read [the Comprehensive Rules document](https://magic.wizards.com/en/game-info/gameplay/rules-and-formats/rules). I am using it as a source for my specs.

I would also suggest you check how I am trying to spec the rules. (My hypothesis is that without proper specs, I am sure to hit the wall quite soon, rather than later.)

**Comment by allentiak** [comment]
This is also the reason why I am starting with defining my Entities (according to "Uncle Bob" Martin's ["Clean Architecture"](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html)) or Data (according to [HTDP's recipe](https://htdp.org/2020-5-6/Book/part_preface.html)).
(You will find more pointers in [magicum's README](https://github.com/allentiak/magicum/blob/master/README.md).)

I suggest you start reading both books. They teach the basics of (respectively) Clean Software Architectures and Systematic Design Process, and they will be much better at explaining these things in detail than me :-)

In other words, I think you have put the cart before the horses, and are now trying to move it forward - I doubt it will move much :-)

**Comment by antonioalonzi** [comment]
Noting to do.
